why-conventional-commits.qmd

```diff
@@ -0,0 +1,278 @@
+
+In general, a commit message should be **concise** (brief, but
+descriptive), **relevant** (focus on *why* the change was necessary, not
+*how* it was implemented), **clear** (include specific descriptions, and
+avoid vagueness and ambiguity), and **consistent** (follow a standard
+format).
+
+Some of the general principles that most conventions follow are that a
+commit message should:
+
+-   Include a short descriptive title, potentially followed by a body
+    with more details.
+-   Use the imperative, present tense in the title and the body (e.g.,
+    "Add feature" instead of "Added feature")
+-   Use the title to summarise the changes made in the commit.
+-   Use the body to provide context and justifications for the change
+    (such as elaboration on the changes, breaking changes, reference of
+    issues, and tagging relevant people).
+:::
+
+## Considered options
+
+::: content-hidden
+List and describe some of the options, as well as some of the benefits
+and drawbacks for each option.
+:::
+
+There are multiple conventions for writing commit messages, but we will
+consider three of the most popular ones: the Angular convention,
+conventional commits, and the Gitmoji convention.
+
+### The Angular convention
+
+The [Angular
+convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines)
+includes a **header**, a **body**, and a **footer** in each commit
+message, like shown below:
+
+```         
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The header includes a **type**, a **scope**, and a **subject**, the body
+includes a **longer description** of the change (including the
+motivation of why the changes was implemented and how it contrasts with
+previous implementations), and the footer includes **metadata** about
+the change (including potential breaking changes and references to
+relevant issues). In the header, the scope is optional, and both the
+body and footer are optional.
+
+This convention originated from the
+[Angular](https://github.com/angular/angular/tree/main) project, which
+is built on TypeScript. It defines specific types for commits, such as
+`feat`, `fix`, `docs`, `style`, `refactor`, `test`, and more, helping to
+categorize changes effectively. In addition, the angular convention
+encourages the use of scope, which should be the name of the npm package
+affected (which is JavaScript-specific) or anything specifying where in
+the codebase the commit change is made.
+
+::: columns
+::: column
+#### Benefits
+
+-   Provides a clear structure for commit messages.
+-   Can be used to generate changelogs (including the omission of
+    irrelevant commits, like formatting).
+:::
+
+::: column
+#### Drawbacks
+
+-   It can be too verbose for some projects.
+-   Usage more prevalent in Angular-related projects.
+-   Developed in a Typescript context
+-   No emojis which can make commit messages easier to scan and more
```

Comment:
It isn't completely clear to me if the emojis makes it easier or more difficult to scan the message :-)

Reply:
I have slightly rewritten this. I hope that it's clearer now :)